### PR TITLE
Add clearer error messages on relation building

### LINKED ISSUE #7828: Misleading errormessage "Something went wrong in the model `X` with the attribute `Y`"
# Setup
I have to models.
- Page
- Shop

They have a many to many relation to each other.
Each Shop can have multiple pages and each page can be used in multiple shops.

I accidentally removed the `collection`field of the `pages` attribute in `shops` Model
```
//shop.settings.json
{
  ...
  "attributes": {
    "pages": {
      "via": "shops",
      "dominant": true
    }
    ...
  }
}
```

```
//page.settings.json
{
  ...
  "attributes": {
    "shops": {
      "collection": "shop",
      "via": "pages",
    }
    ...
  }
}
```

The error message stated: 
"Something went wrong in the model `Page` with the attribute `shops`"

So I expected that `page.settings.json` has an issue.
But that was not the case.
Luckily the documentation https://strapi.io/documentation/v3.x/concepts/models.html#relations showed me that both attributes needed a `collection` property.

# I expect
 the error message to be the other way round
--> "Something went wrong in the model `Shop` with the attribute `pages`"

### PR DESCRIPTION
**Pull request by diogotcorreia**
Fixes #7828 

#### Description of what you did:

I've made Strapi throw an error with a custom message when the attribute of a relation is not correctly configured. This should make error messages more accurate as we can see on #7828.

Using the setup from #7828:
Before this PR: `Something went wrong in the model Page with the attribute shops`
After this PR: `The attribute 'pages' is not correctly configured in the model Shop`

Additionally, I've also added the same error messages on relations using the `model` property.

### CODE REVIEW

**Comment by derrickmehaffy** [comment]
Thank you @diogotcorreia I think this will be much clearer, but I'll let the backend team review

**Review by derrickmehaffy — APPROVED** [review]
LGTM

**Review by Convly — CHANGES_REQUESTED** [review]
Nice addition, thanks!

One thing though, while I was testing I noticed that these 3 exceptions doesn't address every issues on relation building.

In order to be consistent, would you mind add the remaining ones?
Here are a few I've spotted while testing that you can begin with:
- L.78/120: `attribute.collection` or `attribute.model` referring to an unknown model (throws anyway when we try to access the "`resolvedAttribute`" via `models[attribute.xxx]`, but with a generic message)
- L.193: A final else in the case where you remove the collection from the dominant side of a N-N relation. (`attributeName` not configured correctly in `modelName`)

Thanks again for your contribution!

File: packages/strapi-utils/lib/models.js
Comment (Convly) [inline packages/strapi-utils/lib/models.js:135]: Same as above

**Review by Convly — APPROVED** [review]
LGTM, waiting for the tests
_Dismissing the previous review_

File: packages/strapi-utils/lib/models.js
Comment (diogotcorreia) [reply packages/strapi-utils/lib/models.js:84]: I don't think you can get the plugin from which `modelName` is from that function, at least from what I understand. You could get from the other ones because it was specified in the `attributes` variable.

**Comment by Convly** [comment]
Hello @diogotcorreia, there's conflicts with `master` on your branch which prevents the merge. Could you take the time to fix them before it's merged please? :slightly_smiling_face:

**Pull request merged**
